src/components/Verse/VerseSettingsMenu.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import Verse from '../../../types/Verse';
+import VerseSettingsMenuItem from './VerseSettingsMenuItem';
+import CopyIcon from '../../../public/icons/copy.svg';
+
+interface Props {
+  verse: Verse;
+}
+
+const VerseSettingsMenu: React.FC<Props> = ({ verse }) => {
+  const [copied, setCopied] = useState(false);
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+    // if the user has just copied the text, we should change the text back to Copy after 3 seconds.
+    if (copied === true) {
+      timeoutId = setTimeout(() => setCopied(false), 3 * 1000);
+    }
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [copied]);
+
+  const onCopyClicked = () => {
+    navigator.clipboard
```

Comment:
1) Is navigator a browser object? If so, we'd need to wrap it around a browser check. Since next JS runs on the server as well.

2) It looks like Clipboard.writeText doesn't have support on all browsers :/. Is there a 3rd party library that we can use instead? https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write

Reply:
We used `copy-to-clipboard` for writing text to clipboard in [v2](https://github.com/quran/quran.com-frontend-v2/blob/c33675c1994cecf8041134ee2209fe1dbcf2badc/app/javascript/controllers/verse_controller.js#L74)

Reply:
There are a few libraries that do polyfilling for copying to clipboard for example: [clipboard-copy](https://github.com/feross/clipboard-copy), [clipboard-polyfill](https://github.com/lgarron/clipboard-polyfill) and [copy-to-clipboard](https://github.com/sudodoki/copy-to-clipboard) but I think we can go with `copy-to-clipboard` since it's already tried and tested. What do you think? @tifa2UP

Reply:
There are a few libraries that do polyfilling for copying to clipboard for example: [clipboard-copy](https://github.com/feross/clipboard-copy), [clipboard-polyfill](https://github.com/lgarron/clipboard-polyfill) and [copy-to-clipboard](https://github.com/sudodoki/copy-to-clipboard) but I think we can go with `copy-to-clipboard` since it's already tried and tested. What do you think? @tifa2UP

Reply:
I just tried `copy-to-clipboard`, it doesn't seem to work neither on Chrome nor on Safari for some reason :/ `clipboard-copy` does work well.